nix/modules/home-manager/default.nix

```diff
@@ -0,0 +1,158 @@
+        description = ''
+          the command to run for the pinnacle user configuration, provided via the pinnacle config toml file to the pinnacle server binary.
+          this defaults to ''${pkgs.pinnacle-config}/bin/pinnacle-config -- you can provide this package via a nixpkgs overlay like:
+
+          ```nix
+            pkgs = import nixpkgs {
+              inherit system;
+              overlays = [
+                inputs.pinnacle.overlays.default
+                (final: prev: {
+                  pinnacle-config = prev.pinnacle.buildRustConfig {
+                    pname = "pinnacle-config";
+                    version = "0.1.0";
+                    src = ./.;
+                  };
+                })
+              ];
+            };
+          ```
+
+          or by setting the package option directly.
+
+          please note that if you're running this home-manager module on a non-NixOS distribution and making use of snowcap, you need to wrap
+          the call to your configuration script/executable in `nixGL` to ensure the fallback to software rendering isn't used --
+          see: https://pinnacle-comp.github.io/pinnacle/getting-started/running#from-source. you should not use `nix run` here, however. instead,
+          make sure the `nixGL` and `nixVulkanIntel` packages are available and invoke each:
+
+          ```nix
+            services.wayland.windowManager.pinnacle.config.execCmd = ["''${pkgs.nixGL}/bin/nixGL" "''${pkgs.nixVulkanIntel}/bin/nixVulkanIntel" "''${pkgs.pinnacle-config}/bin/pinnacle-config"];
+          ```
+        '';
+      };
+    };
+
+    systemd = lib.mkOption {
+      type = lib.types.submodule systemdModule;
+    };
+
+    extraSettings = mkOption {
+      type = lib.types.attrs;
+
+      default = {};
+
+      example = ''
+        ```nix
+          programs.pinnacle.settings = {
+            env = {
+              "MY_ENV_VAR" = "super special env var";
+            };
+        };
+        ```
+      '';
+
+      description = ''
+        the pinnacle.toml configuration settings exposed as a nix attrset -- these are merged with the settings exposed under the `config` attr.
+
+        see: https://pinnacle-comp.github.io/pinnacle/
+      '';
+    };
+
+    mergedSettings = mkOption {
+      internal = true;
+      type = settingsFormat.type;
+      default = {
+        run = cfg.config.execCmd;
+      } // cfg.extraSettings;
+    };
+  };
+
+  config = let
+    configFile = settingsFormat.generate "pinnacle.toml" cfg.mergedSettings;
+  in lib.mkIf cfg.enable {
+    xdg.configFile."pinnacle/pinnacle.toml" = {
+      source = configFile;
+      # TODO: make pinnacle reload config when this file changes without needing to restart the whole graphical session
+      # example: https://github.com/nix-community/home-manager/blob/2b73c2fcca690b6eca4f520179e54ae760f25d4e/modules/services/window-managers/i3-sway/sway.nix#L726
```

Comment:
one quick/easy option to handle this would be to trap SIGHUP in the pinnacle server and reload config when we receive SIGHUP. then we could use `onChange` to run `kill -HUP $PINNACLE_PID`. we'd need to find the pid but that's potentially easier than adding a msg interface to the pinnacle CLI.